main.cpp

```diff
@@ -238,153 +353,306 @@ struct Bird
     void sing();
-    //    3) eat
-    float eat(std::string target); //input what's being eaten, ouput new total weight
-    std::string forage(Progeny progeny);//makes appropriate food available to progeny to eat based on percentToMaturity
+    void eat(std::string target);
+    void forage(Progeny progeny);
 };
 
+void Bird::Progeny::flyTheNest()
+{
+    float growUp = percentToMaturity + satedLevel;
+    if(.80f < percentToMaturity || 1.f <= growUp)
+    {
+        std::cout << "Your Fledgling grew up!";
+    }
+    else if(.60f <= percentToMaturity)
+    {
+        std::cout << "Your Fledgling survived the attempt, but isn't ready to grow up.";
+    }
+    else
+    {
+        std::cout << "Your Fledgling didn't survive the attempt...";
+    }
+    
+}
+
+void Bird::Progeny::eat(float maturePercent, std::string target)
+{
+    std::cout << "Your Fledgling ate a " << target;
+    percentToMaturity += maturePercent;
+}
+
+int Bird::Progeny::chirp(bool parentPresent, float location, int timeOfDay)
+{
+    int safety = 0;
+    if (parentPresent == true)
+    {
+        safety += 45;
+    }
+    if(location >= 40.f)
+    {
+        safety += 20;
+    }
+    if(7 < timeOfDay && timeOfDay < 18)
+    {
+        safety += 35;
+    }
+    return safety;
+}
+
+double Bird::fly(double x, double y)
+{
+    return sqrt(pow(x, 2.) + pow(y, 2.));
+}
+
+void Bird::sing()
+{
+    std::cout << "The bird sings!";
+}
+
+void Bird::eat(std::string target)
+{
+    std::cout << "Your Bird ate a " << target;
+}
+
+void Bird::forage(Progeny progeny)
+{
+    fledgling = progeny;
+    if(fledgling.percentToMaturity <= .2f)
+    {
+        std::cout << "Your fledglingneeds some seed!";
+    }
+    else if(.2f < fledgling.percentToMaturity && fledgling.percentToMaturity <= .6f)
+    {
+        std::cout << "Your fledgling needs a worm!";
+    }
+    else
+    {
+        std::cout << "Your fledgling needs a beetle!";
+    }
+}
+
 struct CPU
 {
-    //5 properties:
-    //    1) manufacturer (std::string)
     std::string manufacturer = "AMD";
-    //    2) temperature (float)
     float temperature = 55.5f;
-    //    3) clock speed (float)
     float clockSpeed = 3.5f;
-    //    4) cores (int)
     int cores = 6;
-    //    5) socket size (std::string)
     std::string socket = "AM4";
-    //3 things it can do:
-    //    1) store data
-    float storeData(); //returns stored values
-    //    2) output results
-    float outputResults(); //returns current values
-    //    3) compute
-    float compute(float x, float y);//returns result of operations on inputs
+
+    float storeData(); 
+    float outputResults(); 
+    float compute(float x, float y);
 };
 
+float CPU::storeData()
+{
+    float data = 1.43667f * 2.44435f * temperature;
+    return data;
+}
+
+float CPU::outputResults()
+{
+    float data = 89.667f * .30741f * temperature;
+    return data;
+}
+
+float CPU::compute(float x, float y)
+{
+    float data = x * y * temperature;
+    return data;
+}
```

Comment:
Skip the local variable declaration. `float data = ...`
just write `return a * b * c;`
- [ ] revise